include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,313 @@
+	STEPPER_FULL_STEP,
+	STEPPER_MICRO_STEP_2,
+	STEPPER_MICRO_STEP_4,
+	STEPPER_MICRO_STEP_8,
+	STEPPER_MICRO_STEP_16,
+	STEPPER_MICRO_STEP_32,
+	STEPPER_MICRO_STEP_64,
+	STEPPER_MICRO_STEP_128,
+	STEPPER_MICRO_STEP_256,
+};
+
+enum stepper_direction {
+	STEPPER_DIRECTION_POSITIVE,
+	STEPPER_DIRECTION_NEGATIVE,
+};
+
+/**
+ * @typedef stepper_enable_t
+ * @brief enable or disable the stepper motor controller.
+ */
+typedef int32_t (*stepper_enable_t)(const struct device *dev, bool enable);
+
+/**
+ * @typedef stepper_move_t
+ * @brief Move the stepper motor by a given number of steps.
+ */
+typedef int32_t (*stepper_move_t)(const struct device *dev, int32_t steps);
+
+/**
+ * @typedef stepper_set_max_velocity_t
+ * @brief Set the max velocity in steps per seconds.
+ */
+typedef int32_t (*stepper_set_max_velocity_t)(const struct device *dev, uint32_t steps_per_second);
+
+/**
+ * @typedef stepper_set_micro_step_res_t
+ * @brief Set the micro-step resolution
+ */
+typedef int32_t (*stepper_set_micro_step_res_t)(const struct device *dev,
+						enum micro_step_resolution resolution);
+
+/**
+ * @typedef stepper_set_actual_position_t
+ * @brief Set the actual a.k.a reference position of the stepper
+ */
+typedef int32_t (*stepper_set_actual_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @typedef stepper_get_actual_position_t
+ * @brief Get the actual a.k.a reference position of the stepper
+ */
+typedef int32_t (*stepper_get_actual_position_t)(const struct device *dev, int32_t *value);
+
+/**
+ * @typedef stepper_set_target_position_t
+ * @brief Set the absolute target position of the stepper
+ */
+typedef int32_t (*stepper_set_target_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @typedef stepper_is_moving_t
+ * @brief Is the target position fo the stepper reached
+ */
+typedef int32_t (*stepper_is_moving_t)(const struct device *dev, bool *is_moving);
+
+/**
+ * @typedef stepper_enable_constant_velocity_mode_t
+ * @brief Enable constant velocity mode for the stepper with a given velocity
+ */
+typedef int32_t (*stepper_enable_constant_velocity_mode_t)(const struct device *dev,
+							   const enum stepper_direction direction,
+							   const uint32_t value);
+
+/**
+ * @brief Stepper Motor Controller API
+ */
+__subsystem struct stepper_api {
+	stepper_enable_t enable;
+	stepper_move_t move;
+	stepper_set_max_velocity_t set_max_velocity;
+	stepper_set_micro_step_res_t set_micro_step_res;
+	stepper_set_actual_position_t set_actual_position;
+	stepper_get_actual_position_t get_actual_position;
+	stepper_set_target_position_t set_target_position;
+	stepper_is_moving_t is_moving;
+	stepper_enable_constant_velocity_mode_t enable_constant_velocity_mode;
+};
+
+/**
+ * @brief Enable or Disable Motor Controller
+ * @param dev pointer to the stepper motor controller instance
+ * @param enable Input enable or disable motor controller
+ * @retval <0 Error during Enabling
+ * @retval  0 Success
```

Comment:
here and other similar instances

```suggestion
 * @brief Enable or Disable Motor Controller
 
 * @param dev pointer to the stepper motor controller instance
 * @param enable Input enable or disable motor controller

  * @retval <0 Error during Enabling
 * @retval  0 Success
```

Reply:
@jilaypandya I think these formatting tweaks are the last remaining review comment.